clickhouse-admin/src/http_entrypoints.rs

```diff
@@ -46,8 +49,26 @@ impl ClickhouseAdminServerApi for ClickhouseAdminServerImpl {
     ) -> Result<HttpResponseCreated<ReplicaConfig>, HttpError> {
         let ctx = rqctx.context();
         let replica_server = body.into_inner();
-        let output =
-            ctx.clickward().generate_server_config(replica_server.settings)?;
+        let current_generation = ctx.generation().await;
+        let incoming_generation = replica_server.generation();
+
+        // If the incoming generation number is lower, then we have a problem.
+        // We should return an error instead of silently skipping the configuration
+        // file generation.
+        if let Some(current) = current_generation {
+            if current > incoming_generation {
+                return Err(HttpError::for_internal_error(
```

Comment:
This doesn't feel like an internal error to me. This is an expected race condition, and so I think we should return a 400 level error instead of a 500 level error. I think 412 is an appropriate error code, even though we are not using etags for a precondition. @davepacheco does that make sense to you?

Reply:
Definitely agreed it's not a 500.  It looks like Sled Agent uses 409 (Conflict) for this and I'd suggest using that for consistency.
https://github.com/oxidecomputer/omicron/blob/1f0c1856cd3f2ed52115be23e09318e3c2f8af8f/sled-agent/src/services.rs#L3418-L3424
https://github.com/oxidecomputer/omicron/blob/1f0c1856cd3f2ed52115be23e09318e3c2f8af8f/sled-agent/src/services.rs#L308-L310